src/datadog/tracer.cpp

```diff
@@ -74,14 +74,11 @@ Tracer::Tracer(const FinalizedTracerConfig& config,
 
     auto rc_listeners = agent_config.remote_configuration_listeners;
     rc_listeners.emplace_back(config_manager_);
-    auto agent =
-        std::make_shared<DatadogAgent>(agent_config, tracer_telemetry_,
-                                       config.logger, signature_, rc_listeners);
+    auto agent = std::make_shared<DatadogAgent>(
+        agent_config, telemetry_, config.logger, signature_, rc_listeners);
     collector_ = agent;
 
-    if (tracer_telemetry_->enabled()) {
-      agent->send_app_started(config.metadata);
-    }
+    telemetry_->send_app_started(config.metadata);
```

Comment:
I understand removing the if to eval the condition directly in `send_telemetry` is less verbose, however I would see two disadvantages:
- With just a quick look at the code, you can think the telemetry is sent anyway
- It is less performant as we might do two functions call to not send the telemetry in the end. In the send_app_started it is negligible but if the telemetry is more used anywhere else it can have a (small) impact.

Reply:
In regards of the change of the 3rd part, this might not be relevant anymore (as the call will do nothing).

Reply:
when the telemetry is disable, every call will be noop. Also in the future, the tracer will not send the app-started event, instead, will add an event that the telemetry module will send whenever it feels is the right time.

Reply:
That's what I was thinking. Thank you for your confirmation. I resolve this comment.